packages/api/src/controllers/task.ts

```diff
@@ -285,20 +285,43 @@ app.post("/:id/status", authorizer({ anyAdmin: true }), async (req, res) => {
   }
   if (task.inputAssetId) {
     const asset = await db.asset.get(task.inputAssetId);
-    if (task.id === asset?.storage?.status?.tasks.pending) {
-      await req.taskScheduler.updateAsset(asset.id, {
-        storage: {
-          ...asset.storage,
+    if (task.type === "export") {
+      if (task.id === asset?.storage?.status?.tasks.pending) {
+        await req.taskScheduler.updateAsset(asset.id, {
+          storage: {
+            ...asset.storage,
+            status: {
+              phase: "processing",
+              progress: doc.progress,
+              tasks: asset.storage.status.tasks,
+            },
+          },
+        });
+      }
+    } else if (task.type === "delete") {
+      await req.taskScheduler.updateAsset(
+        asset.id,
```

Comment:
**nit:** Since you already have the asset object here, you can just pass it as the argument instead of `asset.id` and avoid a redundant query on the DB inside `updateAsset`. Same for the other update above (it was already like that, my bad)